packages/snaps-utils/src/test-utils/snap.ts

```diff
@@ -27,6 +28,16 @@ export const MOCK_SNAP_ID = 'npm:@metamask/example-snap';
 export const MOCK_LOCAL_SNAP_ID = 'local:@metamask/example-snap';
 export const MOCK_ORIGIN = 'example.com';
 
+type GetPersistedSnapObjectOptions = Partial<Omit<PersistedSnap, 'version'>> & {
+  version?: string | SemVerVersion;
```

Comment:
Why use partial & omit here (and in the other instances that you do this) when you can just change the original type to be SemVerVersion? Also I don't understand why we would want it to be union type w/ string? I thought the whole point of the opaque types was to narrow types?

Reply:
There's tens if not hundreds of tests that use those functions, and we would have to update all of them to have a type cast.

This is for backwards compatibility because of that.

The whole source code (not tests) is type cast though.

Reply:
Also can't use that function, because creation utils take flat list of arguments instead of hierarchy.

Eg
```typescript
export const getSnapManifest = ({
  shasum = DEFAULT_SNAP_SHASUM,
}: GetSnapManifestOptions = {}): SnapManifest => {
  return {
    source: {
      shasum,
    },
  };
};